src/plans/D3MCompoundPlan.sol

```diff
@@ -0,0 +1,198 @@
+//
+// util > kink:
+//         normalRate = kink * multiplierPerBlock / WAD + baseRatePerBlock;
+//         targetInterestRate = normalRate + jumpMultiplierPerBlock * (util - kink) / WAD
+//         util * jumpMultiplierPerBlock = (targetInterestRate - normalRate) * WAD + kink * jumpMultiplierPerBlock
+// (1)     util = kink + (targetInterestRate - normalRate) * WAD / jumpMultiplierPerBlock
+//
+// util <= kink:
+//         targetInterestRate = util * multiplierPerBlock / WAD + baseRatePerBlock
+// (2)     util = (targetInterestRate - baseRatePerBlock) * WAD / multiplierPerBlock
+//
+// https://github.com/compound-finance/compound-protocol/blob/a3214f67b73310d547e00fc578e8355911c9d376/contracts/BaseJumpRateModelV2.sol#L80
+//
+//         util = borrows * WAD / (cash + borrows - reserves);
+// (3)     cash + borrows - reserves = borrows * WAD / util
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPlan.sol";
+
+// cDai - https://etherscan.io/address/0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643
+interface CErc20Like {
+    function totalBorrows()           external view returns (uint256);
+    function totalReserves()          external view returns (uint256);
+    function interestRateModel()      external view returns (address);
+    function getCash()                external view returns (uint256);
+    function implementation()         external view returns (address);
+}
+
+// JumpRateModelV2 - https://etherscan.io/address/0xfb564da37b41b2f6b6edcc3e56fbf523bd9f2012
+interface InterestRateModelLike {
+    function baseRatePerBlock()       external view returns (uint256);
+    function kink()                   external view returns (uint256);
+    function multiplierPerBlock()     external view returns (uint256);
+    function jumpMultiplierPerBlock() external view returns (uint256);
+}
+
+contract D3MCompoundPlan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    InterestRateModelLike        public tack;
+    address                      public delegate; // cDai implementation
+    uint256                      public barb;     // target Interest Rate Per Block [wad] (0)
+
+    CErc20Like public immutable cDai;
+
+    // https://github.com/compound-finance/compound-protocol/blob/a3214f67b73310d547e00fc578e8355911c9d376/contracts/CTokenInterfaces.sol#L31
+    uint256 internal constant MAX_BORROW_RATE = 0.0005e16;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(address cDai_) {
+        address rateModel_ = CErc20Like(cDai_).interestRateModel();
+        address delegate_  = CErc20Like(cDai_).implementation();
+
+        require(rateModel_ != address(0), "D3MCompoundPlan/invalid-rateModel");
+        require(delegate_  != address(0), "D3MCompoundPlan/invalid-delegate");
+
+        cDai     = CErc20Like(cDai_);
```

Comment:
Consider moving this line at the top to avoid redundant cast, following compound pool pattern.

Reply:
Done in `cdd0a82254`